test/Network/MQTT/ParserEncoderSpec.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE FlexibleInstances, MultiParamTypeClasses #-}
+
+module Network.MQTT.ParserEncoderSpec       (spec) where
+
+import           Network.MQTT.Packet
+import           Network.MQTT.Encoder       (encodePacket)
+import           Network.MQTT.Parser        (parsePacket)
+
+import           Test.SmallCheck.Series
+import           Test.Hspec
+import           Test.Hspec.SmallCheck      (property)
+
+import           Data.Attoparsec.ByteString (parseOnly)
+import           Data.Word                  (Word16)
+import           Data.ByteString            (ByteString, singleton)
+import           Data.ByteString.Builder    (toLazyByteString)
+import           Data.ByteString.Lazy       (toStrict)
+import qualified Data.Text as T
+
+spec :: Spec
+spec = do
+  describe "Parser-Encoder" $ do
+    it "test identity" $ property $
+      \p -> let encodedPacket = toStrict $ toLazyByteString (encodePacket p)
+                (Right decodedPacket) = parseOnly parsePacket encodedPacket
+            in decodedPacket == (p :: Packet)
+
+instance Monad m => Serial m Word16 where
+  series = generate $ \d -> take (d+1) [ 0..65535 ]
+
+instance Monad m => Serial m ByteString where
+  series = generate $ \d -> take (d+1) (singleton <$> [ 0..255 ])
```

Comment:
The general note for `smallcheck-series`: we should either use all instances provided (`Word16`, `ByteString`, `Text`) or none at all. We can't select imported instance and they will clash if we implement our own instances.